test/TestConfig/src/org/openj9/envInfo/JavaInfo.java

```diff
@@ -0,0 +1,114 @@
+        String osArch = System.getProperty("os.arch").toLowerCase();
+        System.out.println("System.getProperty('os.arch')=" + System.getProperty("os.arch") + "\n");
+        String cmprssptrs = System.getProperty("java.fullversion");
+        if (cmprssptrs != null) {
+            System.out.println("System.getProperty('java.fullversion')=" + System.getProperty("java.fullversion") + "\n");
+        } else {
+            System.out.println("System.getProperty('java.fullversion')=null\n");
+        }
+        String spec = "";
+        if (osName.contains("linux")) {
+            spec = "linux";
+        } else if (osName.contains("win")) {
+            spec = "win";
+        } else if (osName.contains("mac")) {
+            spec = "osx";
+        } else if (osName.contains("nix") || osName.contains("nux") || osName.contains("aix")) {
+            spec = "aix";
+        } else if (osName.contains("z/os")) {
+            spec = "zos";
+        } else {
+            spec = osName;
+        }
+
+        if (osArch.contains("amd64") || osArch.contains("x86")) {
+            spec += "_x86";
+        } else if (osArch.contains("ppc") || osArch.contains("powerpc")) {
+            spec += "_ppc";
+        } else if (osArch.contains("s390x")) {
+            spec += "_390";
+        } else if (osArch.contains("aarch")) {
+            spec += "_arm";
+        } else {
+            spec += osArch;
+        }
+
+        if (Integer.parseInt(System.getProperty("sun.arch.data.model").trim())==64) {
+            spec += "-64";
+        }
+        System.out.println("System.getProperty('sun.arch.data.model')=" + System.getProperty("sun.arch.data.model") + "\n");
+
+        if (cmprssptrs != null && cmprssptrs.contains("Compressed References")) {
+            spec += "_cmprssptrs";
+        }
+        if (spec.contains("ppc") && osArch.contains("le")) {
+            spec += "_le";
+        }
+        
+        return spec;
+    }
+
+    public int getJDKVersion() {
+        String javaVersion = System.getProperty("java.version");
+        if (javaVersion.startsWith("1.")) {
+            javaVersion = javaVersion.substring(2);
+        }
+        int dotIndex = javaVersion.indexOf('.');
+        int dashIndex = javaVersion.indexOf('-');
+        return Integer.parseInt(javaVersion.substring(0, dotIndex > -1 ? dotIndex : dashIndex > -1 ? dashIndex : javaVersion.length()));
+    }
+
+
+    public String getJDKImpl() {
+        String impl = System.getProperty("java.vm.vendor").toLowerCase();
+        System.out.println("System.getProperty('java.vm.vendor')=" + impl + "\n");
+        try {
+            if (impl.contains("ibm")) {
+                return "ibm";
+            } else if (impl.contains("openj9")) {
+                return "openj9";
+            } else if (impl.contains("oracle") || impl.contains("hotspot") || impl.contains("adoptopenjdk")) {
+                return "hotspot";
+            } else {
+                throw new Exception();
```

Comment:
It is not necessary to throw the exception and try to catch it within the same function. I would recommend returning null here.